packages/autocomplete-js/src/__tests__/detached.test.ts

```diff
@@ -111,10 +111,32 @@ describe('detached', () => {
       '.aa-DetachedCancelButton'
     );
 
+    // Prevent `onTouchStart` event from closing detached overlay
+    const windowTouchStartListener = jest.fn();
+    window.addEventListener('TouchStart', windowTouchStartListener);
```

Comment:
It should be `touchStart` everywhere, right?

```suggestion
    window.addEventListener('touchStart', windowTouchStartListener);
```

Reply:
It should be indeed. I left it with a capital `T` because i wrongly assumed there was an issue with the testing library, but I looked at it again and found the correct source of the issue.

In `setProperty()`, events names are correctly set when they have a global event alias (`on<something>`). But touch events [don't have these currently](https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/ontouchstart#browser_compatibility), so it registers an event listener with the wrong case (`TouchStart` instead of `touchStart` for instance).

I forced the initial in lowercase to fix that, although I understand it would affect custom events names as well, but I don't know if that matter or not in this situation. I could fine-tune it to only remove the initial capital when the name starts with `Touch` if necessary). Wdyt?

Reply:
isn't it meant to be `touchstart`? https://developer.mozilla.org/en-US/docs/Web/API/Document/touchstart_event I'm guessing it happens to work because it's case insensitive?

Reply:
Adding to the confusion is that the previous implementation (`TouchStart`) worked in the browser as well, but not in unit tests. Maybe the browser is more lenient regarding event naming case-sensitivity, whereas the test runner environment is more strict for that matter.

I'll change it to all lowercase, thanks for pointing it out!